## **Défi**

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('datascience_salaries.csv')

In [3]:
df.head()

,Unnamed: 0,job_title,job_type,experience_level,location,salary_currency,salary
0,0,Data scientist,Full Time,Senior,New York City,USD,149000
1,2,Data scientist,Full Time,Senior,Boston,USD,120000
2,3,Data scientist,Full Time,Senior,London,USD,68000
3,4,Data scientist,Full Time,Senior,Boston,USD,120000
4,5,Data scientist,Full Time,Senior,New York City,USD,149000


In [4]:
from sklearn.preprocessing import MinMaxScaler

In [5]:
mms = MinMaxScaler()

In [6]:
df['salary'] = mms.fit_transform(df[['salary']])

In [7]:
df.head()

,Unnamed: 0,job_title,job_type,experience_level,location,salary_currency,salary
0,0,Data scientist,Full Time,Senior,New York City,USD,0.601010
1,2,Data scientist,Full Time,Senior,Boston,USD,0.454545
2,3,Data scientist,Full Time,Senior,London,USD,0.191919
3,4,Data scientist,Full Time,Senior,Boston,USD,0.454545
4,5,Data scientist,Full Time,Senior,New York City,USD,0.601010


In [8]:
from sklearn.decomposition import PCA

Vu que le PCA fonctionne sur des données numériques, on va scinder nos données

In [9]:
df_num = df.select_dtypes(include= ['int64','float64'])

In [10]:
df_num.head()

,Unnamed: 0,salary
0,0,0.601010
1,2,0.454545
2,3,0.191919
3,4,0.454545
4,5,0.601010


In [11]:
# vérifions s'il y a des valeurs manquantes et traitons les
df_num.isnull().sum()

,0
Unnamed: 0,0
salary,0


Il n'y a pas de valeurs manquantes donc on peut passer à l'étape suivante:

Application du PCA

In [12]:
pca = PCA(n_components=2)

In [13]:

p_composantes = pca.fit_transform(df_num)


In [14]:
p_composantes

array([[-9.31620836e+02,  4.27552789e-01],
       [-9.29620836e+02,  2.81082814e-01],
       [-9.28620837e+02,  1.84538865e-02],
       ...,
       [ 1.31837916e+03,  5.28708726e-02],
       [ 1.32337916e+03,  1.24535099e-02],
       [ 1.32737916e+03,  3.50826690e-01]])

In [15]:
#Créons un nouveau Dataframe pour stocker nos composantes :
p_composante_dataset = pd.DataFrame(p_composantes, columns=['PC1','PC2'])

In [16]:
p_composante_dataset.head()

,PC1,PC2
0,-931.620836,0.427553
1,-929.620836,0.281083
2,-928.620837,0.018454
3,-927.620836,0.281077
4,-926.620836,0.427539


In [17]:
#Regroupons l'ensemble de données par la colonne « niveau_d'expérience » et calculez le salaire moyen et médian pour chaque niveau d'expérience (par exemple, junior, intermédiaire, senior).

In [18]:
df.groupby('experience_level')['salary'].mean()

,salary
experience_level,
Entry,0.030864
Executive,0.232712
Mid,0.110035
Senior,0.227717


On remarque que les seniors et les cadres (executive) ont un salaire moyen supérieur à celle des débutants et des intermédiaires.

In [19]:
df.groupby('experience_level')['salary'].median()

,salary
experience_level,
Entry,0.000000
Executive,0.080808
Mid,0.106061
Senior,0.191919


Pour les débutants, la médianne se situe autours de 0  
Pour les Cadre la médiane se situe autour de
0.8,  
Les intermédiaires eux on une médiane de 0.11   
Les seniors ont une médiane de 0.19  
On peut en conclure que le salaire évolue avait dans le temps en fonction du temps
